src/colmap/estimators/cost_functions.h

```diff
@@ -453,6 +453,47 @@ struct MetricRelativePoseErrorCostFunction {
   const EigenMatrix6d sqrt_information_j_;
 };
 
+// Cost function for aligning one 3D point with a reference 3D point with
+// covariance. Convention is similar to colmap::Sim3d
+// r = scale * R * point + t - ref_point
+struct Point3dAlignmentCostFunction {
```

Comment:
nit: I recommend to be more explicit about the coordinate frames, for example, use the following variable names: `b_from_a`, `ref_point_in_b`, `cov_in_b`, etc.